salt/states/module.py

```diff
@@ -658,31 +658,29 @@ def _legacy_run(name, **kwargs):
         if kwargs["returner"] in returners:
             returners[kwargs["returner"]](ret_ret)
     ret["comment"] = f"Module function {name} executed"
-    ret["result"] = _get_result(mret, ret["changes"])
+    ret["result"] = _get_result(mret)
 
     return ret
 
 
-def _get_result(func_ret, changes):
+def _get_result(func_ret):
     res = True
     # if mret is a dict and there is retcode and its non-zero
     if isinstance(func_ret, dict) and func_ret.get("retcode", 0) != 0:
         res = False
         # if its a boolean, return that as the result
     elif isinstance(func_ret, bool):
         res = func_ret
-    else:
-        changes_ret = changes.get("ret", {})
-        if isinstance(changes_ret, dict):
-            if isinstance(changes_ret.get("result", {}), bool):
-                res = changes_ret.get("result", {})
-            elif changes_ret.get("retcode", 0) != 0:
-                res = False
-            # Explore dict in depth to determine if there is a
-            # 'result' key set to False which sets the global
-            # state result.
-            else:
-                res = _get_dict_result(changes_ret)
+    elif isinstance(func_ret, dict):
+        if isinstance(func_ret.get("result", {}), bool):
+            res = func_ret.get("result", {})
+        elif func_ret.get("retcode", 0) != 0:
```

Comment:
This case is already handled on line 699 https://github.com/saltstack/salt/blob/1c23b3dca1c03f374e9024d2aae303cafdb69fba/salt/states/module.py#L669

Reply:
You are right, I missed that while changing it from `changes_ret` to `func_ret`. I will edit that!